File: src/11.c
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size()-1;
        int max = 0;
        while(left < right){
            int temp = min(height[left], height[right])*(right - left);
            if(temp > max)max = temp;
            if(height[left] < height[right])left++;
            else right --;
        }
        return max;
    }
};

File: go/11.go
//Container With Most Water
package main

func maxArea(height []int) int {
	left := 0
	right := len(height) - 1
	var max, tall, area = 0, 0, 0
	for left < right {

		if height[left] > height[right] {
			tall = height[right]
		} else {
			tall = height[left]
		}
		area = tall * (right - left)

		if max < area {
			max = area
		}

		if tall == height[left] {
			left++
		} else {
			right--
		}
	}
	return max
}
